### PR TITLE
Update MenuButton to use new V1 Button

### PR DESCRIPTION
**Pull request by lenahong**
### Platforms Impacted
- [ ] iOS
- [X] macOS
- [X] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes
In the process of updating an interactive-hook, MenuButton's Button type was updated to use the new V1 Button. Because the content and icon related props were renamed and are rendered differently, the approriate changes were made to the MenuButton component.

MenuButton now accepts a String as a child instead of the content prop, and by default will have iconLocation set to 'before' since a chevron takes place after the label. Menu Button now supports V1 Button's new props such as iconOnly

### Verification
menuButtontester

### Pull request checklist

This PR has considered (when applicable):
- [X] Automated Tests
- [X] Documentation and examples
- [X] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Screen Shot 2022-03-30 at 10 29 50 AM

This change causes a redbox on macOS =/ , because MenuButton.macos.tsx is a separate implementation and the child text doesn't pass properly :/ Not sure what's happening off the top of my head

**Comment by rurikoaraki** [comment]
Changing the API surface of the v0 menu button would be a breaking change and probably wouldn't be great for anyone currently using the v0 menu button.

1. I would expect this change on the experimental/V1 Menu Button, is there a reason why we need it on the v0 one?
2. If we do need to make this change on the v0 menu button, can we keep the original API surface (aka takes a content prop and not Text as a child)?

**Review by rurikoaraki — CHANGES_REQUESTED** [review]
See comment

**Comment by lenahong** [comment]
> Changing the API surface of the v0 menu button would be a breaking change and probably wouldn't be great for anyone currently using the v0 menu button.
> 
> 1. I would expect this change on the experimental/V1 Menu Button, is there a reason why we need it on the v0 one?
> 2. If we do need to make this change on the v0 menu button, can we keep the original API surface (aka takes a content prop and not Text as a child)?

Yeah I can keep the original API for menuButton. @Saadnajmi I'll commit this change and let you know so you can test it on macos to see if that fixes it?

**Comment by rurikoaraki** [comment]
@lenahong I'm still unclear on why we need to update v0 MenuButton to use the new Button? Do you mind explaining that?

**Comment by lenahong** [comment]
When I was checking in changes for Android, I needed to change the default useViewCommandFocus hook. @***@***.***> can also chime in here but in the process, MenuButton kept returning an odd error regarding Invalid String range? It went away when I updated the Button type to the new version.

If this update is a concern, I can go back and see if I can revert back the interactive hook and address it like that.

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rurikoaraki** [comment]
> When I was checking in changes for Android, I needed to change the default useViewCommandFocus hook. @***@***.***> can also chime in here but in the process, MenuButton kept returning an odd error regarding Invalid String range? It went away when I updated the Button type to the new version. If this update is a concern, I can go back and see if I can revert back the interactive hook and address it like that. Sent from my T-Mobile 5G Device Get Outlook for Android<https://aka.ms/AAb9ysg>
> […](#)
> ________________________________ From: Ruriko Araki ***@***.***> Sent: Saturday, April 2, 2022 4:01:18 AM To: microsoft/fluentui-react-native ***@***.***> Cc: Lena Hong (SHE/HER) ***@***.***>; Mention ***@***.***> Subject: Re: [microsoft/fluentui-react-native] Update MenuButton to use new V1 Button (PR #1549) @lenahong Lena Hong (SHE/HER) FTE<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Flenahong&data=05%7C01%7CLena.Hong%40microsoft.com%7C0bb3dcdb03e243c9028e08da14120103%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637844364808638673%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=i3%2FIx9UKBofAKuwLmC%2Buzj%2Be4YC%2FtJJLyUyq9Teu%2F6Q%3D&reserved=0> I'm still unclear on why we need to update v0 MenuButton to use the new Button? Do you mind explaining that? — Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Ffluentui-react-native%2Fpull%2F1549%23issuecomment-1086231895&data=05%7C01%7CLena.Hong%40microsoft.com%7C0bb3dcdb03e243c9028e08da14120103%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637844364808638673%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=GIh8L5IksRR6ifhkmGCacSN5s24J%2FPB5RMTf9qRXoIo%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAC766J2IQATZ4WLPGA3O3ETVC5BX5ANCNFSM5SBCBZXQ&data=05%7C01%7CLena.Hong%40microsoft.com%7C0bb3dcdb03e243c9028e08da14120103%7C72f988bf86f141af91ab2d7cd011db47%7C0%7C0%7C637844364808688674%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=MkiYuh0M1Nzsu6P0XDn4DPsfDuJCpz572E6vEk%2FcFcw%3D&reserved=0>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Hmm interesting. I'm worried that this update will cause breaks downstream (for example there's a use of MenuButton in fabric internal which depends on props that are being changed.) I'm okay with us swapping to using the new Button as long as we make sure that the MenuButton props don't change and can be mapped properly to the underlying Button properly (i.e. startIcon is a prop on MenuButton but won't be if we make this change without adding logic to pass startIcon to Button's icon & iconPosition props)

File: packages/components/MenuButton/src/MenuButton.types.ts [hunk 1]
```diff
@@ -24,7 +25,8 @@ export interface MenuButtonItemProps extends ContextualMenuItemProps {
   showSubmenu?: boolean;
 }
 
-export interface MenuButtonProps extends ButtonProps {
+export interface MenuButtonProps extends Omit<ButtonProps, 'icon'> {
+  startIcon?: IconSourcesType;
```
Comment (rurikoaraki) [inline packages/components/MenuButton/src/MenuButton.types.ts:29]: The old button also allowed for `endIcon`, could we add that too?

This page has a full diff of the props between old and new button: https://github.com/microsoft/fluentui-react-native/blob/master/packages/components/Button/MIGRATION.md#props-no-longer-supported-with-an-equivalent-functionality-in-v1-button

**Review by rurikoaraki — CHANGES_REQUESTED** [review]
See comment

**Review by rurikoaraki — APPROVED** [review]

**Comment by Saadnajmi** [comment]
.. well it does run without issues on macOS so we're good!

**Pull request merged**